docs/helm-deployment-aws_kops.md

```diff
@@ -0,0 +1,702 @@
+
+| Component   | Getting Started Guide |
+| ------------| --------------------- |
+| Docker      | https://docs.docker.com/ |
+| Kubernetes  | https://kubernetes.io/docs/tutorials/kubernetes-basics/ |
+| Kubectl     | https://kubernetes.io/docs/tasks/tools/install-kubectl/ |
+| Helm        | https://docs.helm.sh/using_helm/#quickstart-guide |
+| Kops        | https://github.com/kubernetes/kops/blob/master/docs/aws.md |
+| AWS Cli     | https://github.com/aws/aws-cli#installation |
+
+**Note:** You don't need to clone this repository to deploy Alfresco Content Services Community.
+
+### Prepare and configure Kops
+
+Kops requires certain configurations on your AWS account. Please follow the Kops guide to
+[set up your environment](https://github.com/kubernetes/kops/blob/master/docs/aws.md#setup-your-environment)
+through to "Creating your first cluster". The cluster for ACS will be created in the next step.
+
+### Setting up Kubernetes cluster on AWS with Kops
+
+* Export the AWS user credentials:
+```bash
+# Access Key & Secret
+export AWS_ACCESS_KEY_ID=XXXXXXXXXXXXXXXXXXXXXXXX
+export AWS_SECRET_ACCESS_KEY=XXXXXXXXXXXXXXXXXXXXXXXX
+(or)
+# AWS Profile
+export AWS_PROFILE="<profile-name>"
+
+export AWS_DEFAULT_REGION="<region-name>"
+```
+
+* Name the Alfresco Content Services Community environment:
+```bash
+export KOPS_NAME="myacs.example.com"
+```
+
+* Provide an S3 bucket name to store the cluster configuration used by `kops`.  This bucket needs to be created first:
+```bash
+# If s3 bucket is not already created
+$ aws s3 mb s3://my-bucket-kops-store
+
+export KOPS_STATE_STORE="s3://my-bucket-kops-store"
+```
+
+* Create a Kubernetes cluster using the `kops` utility:
+```bash
+kops create cluster \
+  --ssh-public-key <SSH-PUBLIC-KEY> \
+  --name $KOPS_NAME \
+  --state $KOPS_STATE_STORE \
+  --node-count 2 \
+  --zones <AWS-REGION_AVAILABILITY-ZONE(s)> \
+  --master-zones <AWS-REGION_AVAILABILITY-ZONE(s)> \
+  --cloud aws \
+  --node-size m4.xlarge \
+  --master-size t2.medium \
+  -v 10 \
+  --kubernetes-version "1.11.6" \
+  --bastion \
+  --topology private \
+  --networking weave \
+  --yes
+```
+Adjust the above values accordingly (ex: `--node-size`, `--kubernetes-version` etc.).
+**Note:** The default bastion user name is `admin` accessible via the Bastion ELB (and not the EC2 host where the bastion is running).
+
+### Validate cluster
+
+It is important to confirm the status of the Kops creation of a Kubernetes cluster before it can be utilised.
+
+```bash
+kops get cluster
+kops validate cluster
+```
+
+### Upgrading a cluster (optional)
+
+* Edit the worker nodes configuration to include spot price:
+```
+kops edit ig nodes
+```
+
+Include the following in the configuration:
+```
+spec:
+  maxPrice: "0.07"
+```
+
+* (Optional) Enable the required admission controllers, for example, by applying a [Pod Security Policy](k8s-pod-security-policies.md).
```

Comment:
URL not working from feature/DOCS-5309-Update-docs branch, is this supposed to work on master branch ?